### ISSUE TITLE
Add Event for non-legacy browsers with `onlyLegacy: true`

### ISSUE THREAD

**Issue by dwt47** [jhildenbiddle/css-vars-ponyfill#113]
Hi, thank you for building + maintaining such a great ponyfill!

I have some code (initializing a Vue app) that I want to have run as early as possible, but not before the `cssVars` call completes on legacy browsers. I also don't want to have to add `onlyLegacy: false` because I don't want extra work to be done in non-legacy browsers. However, handlers like `onComplete` don't trigger at all in non-legacy browsers, so adding my initialization code to `onComplete` means that it never runs in non-legacy browsers.

Could you add a handler that triggers after `cssVars` call is done, even in non-legacy browsers and with `onlyLegacy: true` set? It could either be a handler called only when `cssVars` bails early or one that's called 100% of the time.

Thank you!

**Comment by jhildenbiddle** [comment]
Thanks for the kind words, @dwt47. Much appreciated.

Check out these two issues: #59, #98. 

With regards to adding a final callback that fires in all browsers, it can be added easily enough but I'm generally hesitant to add yet-another-option/callback unless absolutely necessary. That said, this is the fourth issue filed asking for it (#14 + the two issues linked above), so perhaps it's time to reconsider.

Let me know if either of the approaches in the two linked issues work for you and we'll go from there.

**Comment by dwt47** [comment]
Thanks for the response + your time! Those approaches do the trick, but leave open the possibility of missing out or doubling up on browsers in the future if the criteria used by this package to determine support ever changes (since then "my" way of determining it wouldn't match "yours"). 

I think it's unlikely that this criteria/implementation will ever change, but it still feels much safer to guarantee that the logic handling whether or not to do this processing is the same in all cases.

**Comment by jhildenbiddle** [comment]
Makes sense. Here are my initial thoughts:

* Callback to be named `onDone`
* Last callback invoked by the ponyfill (after `onComplete`, if applicable)
* Invoked in all browsers
* Returns two values: `hasNativeSupport` and `benchmark`

**Examples**

```js
cssVars({
  onDone(hasNativeSupport, benchmark) {
    // All browsers...
    console.log('Benchmark: ' + benchmark);
  }
});
```

The `onComplete` callback will continue to be where you access the CSS output, DOM elements, and custom property key/value pairs processed by the ponyfill, while the `onDone` callback can be used to invoke post-ponyfill code in both modern and legacy browsers:

```js
cssVars({
  onComplete(cssText, styleElms, cssVariables) {
    // Legacy browsers...
  },
  onDone(hasNativeSupport, benchmark) {
    // All browsers...
    console.log('Benchmark: ' + benchmark);
  }
});
```

If you don't need to access the values returned by the `onComplete` callback, you can move all post-ponyfill logic into the `onDone` callback and use the `hasNativeSupport` argument to target modern and/or legacy browsers:

```js
cssVars({
  onDone(hasNativeSupport, benchmark) {
    // All browsers...
    console.log('Benchmark: ' + benchmark);

    // Modern browsers...
    if (hasNativeSupport) {
      console.log('Native Support');
    }
    // Legacy browsers...
    else {
      console.log('Legacy');
    }
  }
});
```

The only bummer is that this will require a major version bump since the `benchmark` value is being moved from the `onComplete` callback to the `onDone` callback.

Thoughts?

Pinging @pseudoramble, @freitasskeeled, and @evry-johber since they created issues related to this same topic earlier.

**Comment by evry-johan** [comment]
@jhildenbiddle Thanks for your time to address this, I think it sounds like just what I asked for. Just as both @dwt47 and I have mentioned, it makes more sense to let the package handle the logic to determine "legacy or not" and provide a single callback, rather than letting the package users copy-paste the logic, which in the end is a type of DRY violation.

**Comment by dwt47** [comment]
This looks great to me! Solves exactly what I wanted it to solve and in a clear, concise way.

Thanks for all your hard work!

**Comment by jhildenbiddle** [comment]
Sounds good to me then. I'll take care of this early next week.

Thanks for the quick feedback, everyone!

**Comment by jhildenbiddle** [comment]
Apologies for the delay on this update, folks. The world got crazy.

I've added a new `onFinally()` callback and made it available on a separate branch: [113](https://github.com/jhildenbiddle/css-vars-ponyfill/tree/113). I wanted to make this new feature available for review to ensure that it meets your needs before merging it into the master branch and pushing a new version to npm.

From the docs:

>### onFinally
>
>- Type: `function`
>- Arguments:
>   1. **hasChanged**: A `boolean` indicating if the last ponyfill call resulted in a style change
>   1. **hasNativeSupport**: A `boolean` indicating if the current browser provides [native support](https://caniuse.com/#feat=css-variables) for CSS custom properties
>   1. **benchmark**: A `number` representing to the ponyfill execution time in milliseconds
>
>Callback after changes have been applied in modern and legacy browsers.
>
>**Example**
>
>```javascript
>cssVars({
>  onFinally: function(hasChanged, hasNativeSupport, benchmark) {
>    // All browsers
>    // ...
>
>    // Modern browsers
>    if (hasNativeSupport) {
>      // ...
>    }
>    // Legacy browsers
>    else {
>      // ...
>    }
>  }
>});
>```

Thanks in advance for any feedback you can provide. Much appreciated.

**Comment by jhildenbiddle** [comment]
New `onFinally` callback available in v2.3.0.
